apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -2755,6 +2758,11 @@ def consolidate_and_vacuum(
         )
         logger.debug("Number of workers %d", workers)
 
+        if distance_metric != vspy.DistanceMetric.L2 and index_type != "FLAT":
```

Comment:
I would suggest to perform this check with the `create` function of each algorithm.